system_intelligence/gpus_info.py

```diff
@@ -0,0 +1,111 @@
+import typing as t
+from rich import print
+
+from .base_info import BaseInfo
+
+
+class QueryError(RuntimeError):
+    """
+    Indicate that a query failed.
+    """
+
+
+compute_capability_to_architecture = {
+    2: 'Fermi',
+    3: 'Kepler',
+    5: 'Maxwell',
+    6: 'Pascal',
+    7: 'Volta',
+    8: 'Ampere'
+}
+
+try:
+    import pycuda
+    import pycuda.driver as cuda
+    import pycuda.autoinit  # noqa F401
+    is_CUDA_available = True
+except (ModuleNotFoundError, ImportError):
+    is_CUDA_available = False
+
+
+class GpusInfo(BaseInfo):
+    """
+    Query any info on GPUs available for the users system (currently only supports NVIDIA GPUs)
+    """
+    def __init__(self):
+        super().__init__()
+
+    def query_gpus(self, **_) -> t.List[t.Mapping[str, t.Any]]:
+        """
+        Get information about all GPUs.
+        """
+        if not is_CUDA_available:
+            print('[bold yellow]Unable to import package pycuda. GPU information may be limited.')
+            return []
+
+        gpus = []
+        for i in range(cuda.Device.count()):
+            device = cuda.Device(i)
+            gpus.append(GpusInfo.query_gpu(device))
+
+        return gpus
+
+    @staticmethod
+    def query_gpu(device: 'cuda.Device') -> t.Mapping[str, t.Any]:
+        """
+        Get information about a given GPU.
+        """
+        attributes = device.get_attributes()
+        compute_capability = device.compute_capability()
+        multiprocessors = attributes[cuda.device_attribute.MULTIPROCESSOR_COUNT]
+        cuda_cores = GpusInfo.calculate_cuda_cores(compute_capability, multiprocessors)
+        try:
+            return {
+                'architecture': compute_capability_to_architecture[compute_capability[0]],
+                'brand': device.name(),
+                'compute_capability': str(float('.'.join(str(_) for _ in compute_capability))),
```

Comment:
That's interesting. It should usually only every be a single compute capability. Mixing different GPU generations works almost never. -> Even when using 2 GPUs the compute capability should only be a single number

Reply:
Yeah, I didn't change this code here (besides the different line number), but I also do not habe the possibility of testing if I remove this (atm)

Reply:
I could test it

Reply:
@Zethson  could pls remove and test it quick?

Reply:
Think I had a wrong thought. This should be all right.